### PR TITLE
Create a nprogress hook

### PR DESCRIPTION
**Pull request by Morpa**
Created a hook for NProgress library to make the root file cleaner.

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
 import nProgressStyles from 'nprogress/nprogress.css'
-import { useEffect } from 'react'
-
 import {
-  Links,
-  LiveReload,
-  Meta,
-  Outlet,
+  Links, LinksFunction, LiveReload,
+  Meta, MetaFunction, Outlet,
```
Comment (willianjusten) [inline app/root.tsx:4]: Is this something coming from your prettier? I don't like having many in the same line like this...

Comment (Morpa) [reply app/root.tsx:4]: Yep, I fix it.

Comment (Morpa) [reply app/root.tsx]: It´s ok now. Sorry about my mistake

Comment (willianjusten) [reply app/root.tsx]: Can you fix the conflicts as well, please? =)

Comment (Morpa) [reply app/root.tsx]: Done.

Comment (Morpa) [reply app/root.tsx]: Do you like the spinner on the corner? Or I can remove?

File: app/root.tsx [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
   Outlet,
   Scripts,
-  ScrollRestoration,
-  useTransition,
+  ScrollRestoration
 } from 'remix'
-import type { LinksFunction, MetaFunction } from 'remix'
 
 import { Layout } from './components/Layout'
+import { useNProgress } from './hooks/use-nprogress'
```
Comment (adeonir) [inline app/root.tsx:14]:
```suggestion
import { Layout } from '~/components/Layout'
import { useNProgress } from '~/hooks/use-nprogress'
```

File: app/root.tsx (hunk 1 above)
Comment (adeonir) [reply app/root.tsx]: @Morpa please merge the latest `main` to your repo. I just pushed a little fix to lint-staged script that runs `eslint` and `prettier`.

Comment (Morpa) [reply app/root.tsx]: @adeonir of course

File: app/root.tsx [hunk 3]
```diff
@@ -82,7 +74,7 @@ export default function App() {
 }
 
 export function ErrorBoundary({ error }: ErrorBoundaryProps) {
-  console.error(error)
+  console.error(error);
```
Comment (helderburato) [inline app/root.tsx:77]: Is this `console.error` expected to be here?

Comment (willianjusten) [reply app/root.tsx:77]: Yep, it's not from his code but from the ErrorBoundary, it's useful to debug situations.

File: app/hooks/use-nprogress.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import nProgress from 'nprogress'
+import { useEffect } from 'react'
+import { useTransition } from 'remix'
+
+/**
+ * When the state is idle then we can to complete the progress bar
+ * and when it's something else it means it's either submitting a form or
+ * waiting for the loaders of the next location so we start it
```
Comment (willianjusten) [inline app/hooks/use-nprogress.ts:8]: Just rephrasing it a little bit (I was probably drunk when I wrote this lol)

```suggestion
 * When the state turns to idle it means the interaction is done.
 * When it's not idle, it means something is happening yet,
 * could be waiting for the loaders of the next location or submitting a form.
```

**Review by willianjusten — APPROVED** [review]
It looks good now, just take my latest iteration on the comments and we're good to go.

**Review by adeonir — APPROVED** [review]

**Pull request merged**